### PR TITLE
Cross compile to SBT 1.0 and 0.13

### PR DESCRIPTION
**Pull request by joan38**
Hi,

Really keen to merge this as soon as possible as we are currently forced to move on SBT 1.0 due to Scala 2.12 requirements.

Cheers

### CODE REVIEW

**Comment by joan38** [comment]
Seems failing because of some 2.10 config on Travis?

**Comment by joan38** [comment]
Build is now green, can anyone review this ASAP?

**Comment by joan38** [comment]
Is this project still alive?

**Comment by joan38** [comment]
Should I fork this project as it seems dead?

**Comment by joan38** [comment]
cc @simon-morgan @margussipria @tpunder @danklynn @rmarsch @er1c ?

**Comment by er1c** [comment]
@joan38 sorry about that!  For some reason I wasn't getting alerts on this.  We can review this and get it committed tomorrow if that is ok?

**Comment by joan38** [comment]
Hi @er1c I was a bit desperate!
That great for tomorrow.
Thanks

File: src/main/scala-sbt-1.0/fm/sbt/CrossSbtUtils.scala [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package fm.sbt
+
+import sbt.librarymanagement.RawRepository
+
+object CrossSbtUtils {
+  def rawRepository(resolver: S3URLResolver, name: String) = new RawRepository(resolver, name)
```
Comment (er1c) [inline src/main/scala-sbt-1.0/fm/sbt/CrossSbtUtils.scala]: I'd be tempted to just make the signature:

`def makeRawRepository(resolver: S3URLResolver): RawRepository = new RawRepository(resolver, resolver.name)`

Is the `CrossSbtUtils` convention something you're seeing in other plugin projects for helper methods?

File: build.sbt [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
-
 version := "0.12.0-SNAPSHOT"
-
 description := "SBT S3 Resolver Plugin"
-
 licenses := Seq("Apache License, Version 2.0" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-
 homepage := Some(url("https://github.com/frugalmechanic/sbt-s3-resolver"))
+sbtPlugin := true
+crossSbtVersions := Seq("0.13.16", "1.0.0")
```
Comment (er1c) [inline build.sbt]: Does this make sense to bump to `1.0.1`?

Comment (joan38) [reply build.sbt]: Yes indeed

**Comment by tpunder** [comment]
Hello, I was out of town for the past week and a half which is why you got no response. 

I started working on SBT 1.0 support a while ago (basically when SBT 1.0 came out) and ran into a few issues with the cross building and sbt-release plugin that I still haven't figured out.

I've committed what I had and have attempted to publish `fm-sbt-s3-resolver` version `0.12.0` which should be compatible with SBT 1.0.  I'll try to get the cross building / cross publishing to work properly this week.

File: src/main/scala-sbt-1.0/fm/sbt/CrossSbtUtils.scala (hunk 1 above)
Comment (joan38) [reply src/main/scala-sbt-1.0/fm/sbt/CrossSbtUtils.scala]: See:
https://github.com/vmunier/sbt-web-scalajs/pull/38
http://www.scala-sbt.org/1.0/docs/Migrating-from-sbt-013x.html#sbt+version+specific+source+directory

**Comment by joan38** [comment]
What is the issue you are hitting?

**Comment by joan38** [comment]
Great I've been able to use 0.12.0 with 1.0.
I'm closing this PR.
Thanks @tpunder

**Comment by tpunder** [comment]
@joan38 Is 0.12.0 working for you?  I'm having issues trying to publish (and probably trying to resolve too).

I think https://github.com/sbt/librarymanagement is interfering with how I hook into Ivy: https://github.com/sbt/librarymanagement/blob/480a1a366ddd69e4179fb097eb84163a1ffbf42f/ivy/src/main/scala/sbt/internal/librarymanagement/Ivy.scala#L91

**Comment by joan38** [comment]
Tes everything works for me with SBT 1.0 and 0.12.0

**Pull request closed without merging**